vortex-metrics/src/lib.rs

```diff
@@ -3,22 +3,34 @@
 
 use std::borrow::Cow;
 use std::collections::BTreeMap;
+use std::fmt::{Debug, Formatter};
 use std::sync::Arc;
 
-use witchcraft_metrics::{Metric, MetricRegistry, Metrics, MetricsIter};
+use parking_lot::Mutex;
+use witchcraft_metrics::{MetricRegistry, Metrics, MetricsIter};
 
 /// A metric registry for various performance metrics.
 #[derive(Default)]
 pub struct VortexMetrics {
     registry: MetricRegistry,
     default_tags: DefaultTags,
+    children: Arc<Mutex<Vec<Arc<VortexMetrics>>>>,
```

Comment:
What are your thoughts on parking_lot? It sounds like parking_lot is better when the lock is not contended, but if it's highly contended then std wins. I've never used it though

Reply:
yea that is my understanding as well, one more difference is parking lot doesn't support lock poisoning but in this case I don't think it matters that much. I have no strong preference though

Reply:
locks not returning results is a huge win IMO 😂